src/server/rdb_save.cc

```diff
@@ -1406,6 +1406,13 @@ RdbSaver::RdbSaver(::io::Sink* sink, SaveMode save_mode, bool align_writes) {
 }
 
 RdbSaver::~RdbSaver() {
+  // Decommit local memory.
+  // We create an RdbSaver for each thread, so each one will Decommit for itself.
+  auto* tlocal = ServerState::tlocal();
+  if (tlocal) {
```

Comment:
when is this null?

Reply:
tests :smile: 

For example `LoadEmpty` in `rdb_test.cc`, does not run `RdbLoad::Load` in a shard thread so it crashes. 

I do not know if a similar unit test exist for `rdb_save` but I assumed this is the case and I added this here as well to avoid crashes like that

Reply:
that odd because rdb_save calls service init. please check this again why it is not set

Reply:
I don't see why this is odd. The service is created but we just don't run the rdb loader on shard thread